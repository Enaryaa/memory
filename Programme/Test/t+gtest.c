#include<stdio.h>
#include<stdlib.h>
#include<graph.h>
#include<math.h>

#define longueurImage 95
#define largeurImage 95
#define CYCLE 1000000L
#define cartex 4
#define cartey 4

void timer(void)
    {
        
        char buf[100];
        int seconde = 0;
 
         ChoisirEcran(1);
         CopierZone(2,1,0,0,800,500,0,0);
         snprintf(buf,100,"temps : %5d",seconde);
         EcrireTexte(((longueurImage*8 + 30*8 + 20)/2),30,buf,1);
         CopierZone(1,0,0,0,800,500,0,0);


    }

int main(void){
	int marge = 30; 
	int x = marge;
	int y = 70; 
	int i, j;
	int goon = 1;
	int seconde = 0;
    unsigned long Microsecondes(), temps;
    temps = (Microsecondes()+CYCLE);

InitialiserGraphique();
CreerFenetre(10,25,800,500);
ChoisirEcran(2);
ChargerImageFond("../../Image/fond.png");

while(goon){

for( i = 0; i < cartex; i++  ){
		for (j = 0; j < cartey; j++){
			int sprite = ChargerSprite("../../Image/carte1.png");
			AfficherSprite(sprite, x, y);
			x = x + marge + longueurImage;
		}
		x = marge;
		y = y + marge + largeurImage;

	

	if (Microsecondes() >temps){
		seconde++;
		timer();
		temps = (Microsecondes() + CYCLE);
	}
	}
}


Touche();
FermerGraphique();

	return EXIT_SUCCESS;
}