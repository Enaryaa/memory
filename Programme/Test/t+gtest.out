#include<stdio.h>
#include<stdlib.h>
#include<graph.h>
#include<math.h>

#define longueurImage 95
#define largeurImage 95
#define CYCLE 1000000L
#define cartex 4
#define cartey 4

void timer(void)
    {
        
        unsigned long Microsecondes(), temps;
        char buf[100];
        int seconde = 0;        /*initialise seconde à 0 pour commencer le temps à partir de 0*/

        InitialiserGraphique();   /*créer la fenetre */
        CreerFenetre(200,190,1280,720);

        temps = Microsecondes() + CYCLE;

        while (goon == 1)
        {
  
            if (Microsecondes()>temps){
            	ChoisirEcran(1);
            	CopierZone(2,1,0,0,800,500,0,0);
                seconde++;                              /* compte le nombre de seconde*/
                temps = (Microsecondes()+CYCLE);
                EcrireTexte(((longueurImage*8 + 30*8 + 20)/2),30,buf,1);
                snprintf(buf,100,"temps : %5d",seconde);
                CopierZone(1,0,0,0,800,500,0,0);
            }
        }

    }

int main(void){
	int marge = 30; 
	int x = marge;
	int y = 70; 
	int i, j;
	int goon = 1;

InitialiserGraphique();
CreerFenetre(10,25,800,500);
ChoisirEcran(2);

timer();
	for( i = 0; i < cartex; i++  ){
		for (j = 0; j < cartey; j++){
			int sprite = ChargerSprite("../../Image/carte1.png");
			AfficherSprite(sprite, x, y);
			x = x + marge + longueurImage;
		}
		x = marge;
		y = y + marge + largeurImage;

	}

Touche();
FermerGraphique();

	return EXIT_SUCCESS;
}